### PR TITLE
support for automation environment and tool data on useragent

### PR DESCRIPTION
**Pull request by benPearce1**
[sc-31593]

### CODE REVIEW

File: pkg/services/api/automation_environment_provider.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"Hudson":       {"HUDSON_URL"},
+	"MagnumCI":     {"MAGNUM"},
+	"SailCI":       {"SAILCI"},
+	"Semaphore":    {"SEMAPHORE"},
+	"Shippable":    {"SHIPPABLE"},
+	"SolanoCI":     {"TDDIUM"},
+	"StriderCD":    {"STRIDER"},
+}
+
+func GetAutomationEnvironment(e Environment) string {
```
Comment (benPearce1) [inline pkg/services/api/automation_environment_provider.go:46]: this is to support mocking the `os.Getenv` function

File: pkg/services/api/api.go [hunk 2]
```diff
@@ -63,5 +64,8 @@ func GetUserAgentString() string {
 			}
 		}
 	}
-	return fmt.Sprintf("%s/%s (%s; %s) go/%s", "go-octopusdeploy", version, runtime.GOOS, runtime.GOARCH, runtime.Version())
+
+	automationEnvironment := GetAutomationEnvironment(&OsEnvironment{})
```
Comment (benPearce1) [inline pkg/services/api/api.go:68]: at runtime we will pass in the real environment

File: pkg/client/octopusdeploy.go [hunk 3]
```diff
@@ -169,6 +169,12 @@ func IsAPIKey(apiKey string) bool {
 // NewClient returns a new Octopus API client. If a nil client is provided, a
 // new http.Client will be used.
 func NewClient(httpClient *http.Client, apiURL *url.URL, apiKey string, spaceID string) (*Client, error) {
+	return NewClientForTool(httpClient, apiURL, apiKey, spaceID, "")
+}
+
+// NewClientForTool returns a new Octopus API client with a tool reference in the useragent string.
+//If a nil client is provided, a new http.Client will be used.
+func NewClientForTool(httpClient *http.Client, apiURL *url.URL, apiKey string, spaceID string, requestingTool string) (*Client, error) {
```
Comment (benPearce1) [inline pkg/client/octopusdeploy.go:177]: an overload to set the requesting tool in the agent string

**Comment by jbristowe** [comment]
Thinking out loud here: Maybe we should consider using `context` instead of parameter passing for the tool information? A good rule of thumb is that any data required for a function to run should be passed as parameters. In this case (however), it's contextual info.

@domenicsim1 thoughts?

**Review by domenicsim1 — APPROVED** [review]

**Pull request merged**